spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BinderAwareChannelResolver.java

```diff
@@ -86,8 +93,23 @@ else if (tokens.length != 1) {
 								" [<transport>:]<name>");
 					}
 				}
-				Binder<MessageChannel> binder = binderFactory.getBinder(transport);
-				binder.bindProducer(name, channel, this.producerProperties);
+				String[] dynamicDestinations = null;
+				Properties producerProperties = null;
+				if (this.channelBindingServiceProperties != null) {
+					dynamicDestinations = this.channelBindingServiceProperties.getDynamicDestinations();
+					// TODO: need the props to return some defaults if not found
+					producerProperties = this.channelBindingServiceProperties.getProducerProperties(name);
+				}
+				boolean dynamicAllowed = dynamicDestinations == null
+						|| dynamicDestinations.length == 0
+						|| Arrays.asList(dynamicDestinations).contains(name);
+				if (dynamicAllowed) {
+					Binder<MessageChannel> binder = binderFactory.getBinder(transport);
+					binder.bindProducer(name, channel, producerProperties);
```

Comment:
While reviewing this, I realized that we never unbind these channels. While the endpoints are in principle stopped eventually, I still think it'd be preferable if the bindings would be part of the managed input/output binding lifecycle.